homeassistant/components/trello/__init__.py

```diff
@@ -0,0 +1,58 @@
+"""The Trello integration."""
+from __future__ import annotations
+
+from trello import Member, TrelloClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+PLATFORMS: list[str] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up from a config entry."""
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    entry.async_on_unload(entry.add_update_listener(async_update_entry))
+
+    return True
+
+
+async def async_update_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Update a given config entry."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    return True
+
+
+async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle removal of an entry."""
+    await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+class TrelloAdapter:
+    """Adapter for Trello lib's client."""
+
+    def __init__(self, client: TrelloClient) -> None:
```

Comment:
Make hass a parameter and do all the `self.hass.async_add_executor_job` in here and have an async API

Reply:
We can discuss further if you feel very strongly about this, but I'm going to keep this as is. The adapter is meant to be a very thin, local client that wraps the 3rd party dependency for the purposes of this integration. Having only the 3rd party client as a dependency keeps it simple and completely unit testable. It also gives us a place to put a well defined interface with the 3rd party client.

Currently there are some other places where we interact with the 3rd party client directly, but eventually I'd like to refactor all usages to go through this adapter instead.